spyder/plugins/editor/lsp/tests/fixtures.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+
+# Copyright © Spyder Project Contributors
+# Licensed under the terms of the MIT License
+# (see spyder/__init__.py for details)
+
+import os
+try:
+    from unittest.mock import Mock
+except ImportError:
+    from mock import Mock # Python 2
+
+from qtpy.QtCore import QObject, Signal, Slot
+import pytest
+from pytestqt.plugin import QtBot
+
+from spyder.config.main import CONF
+from spyder.plugins.editor.lsp import SERVER_CAPABILITES
+from spyder.plugins.editor.lsp.manager import LSPManager
+
+
+class EditorMock(QObject):
+    """
+    Mock for the Editor plugin with the interface needed by LSPManager.
+    """
+    sig_lsp_initialized = Signal()
+
+    def __init__(self):
+        QObject.__init__(self)
+        self.lsp_editor_settings = {}
+
+    @Slot(dict, str)
+    def register_lsp_server_settings(self, settings, language):
+        self.lsp_editor_settings[language] = settings
+        self.sig_lsp_initialized.emit()
+
+
+class MainWindowMock(QObject):
+    """Mock for the Main Window."""
+    def __init__(self):
+        QObject.__init__(self)
+        self.editor = EditorMock()
+
+    def __getattr__(self, attr):
+        if attr == 'editor':
+            return self.editor
+        elif attr == 'projects':
+            # TODO: Add tests for project switching
+            return None
+        else:
+            return Mock()
+
+
+@pytest.fixture(scope="module")
```

Comment:
Is there any reason we need this? seems to only be used on the fixture below?

Reply:
I mean why not add all on `lsp_manager` instead of having `qtbot_module` and `lsp_manager`

Reply:
Because `qtbot` is a `scope=function` fixture, so you can't create a `scope=module` fixture with it (`pytest` rules forbid it). So we have to redefine it as a `module` fixture to be able to catch the Qt signal that tells us that an LSP client is up and ready.

Besides, this fixture is also used by `lsp_codeeditor` now.

Reply:
Ok, thanks for the explanation